src/Microsoft.SqlTools.ServiceLayer/Cms/CmsService.cs

```diff
@@ -0,0 +1,434 @@
+                }
+            }
+            return null;
+        }
+
+        private async Task<ServerConnection> ValidateAndCreateConnection(ConnectParams connectionParams)
+        {
+            // Validate Parameters and Create Connection
+            ConnectionCompleteParams connectionCompleteParams = await ConnectionServiceInstance.Connect(connectionParams);
+            if (!string.IsNullOrEmpty(connectionCompleteParams.Messages))
+            {
+                throw new Exception(connectionCompleteParams.Messages);
+            }
+
+            // Get Connection
+            ConnectionInfo connectionInfo = ConnectionServiceInstance.OwnerToConnectionMap[connectionParams.OwnerUri];
+            ServerConnection serverConn = ConnectionService.OpenServerConnection(connectionInfo);
+
+            return serverConn;
+        }
+
+        private ServerConnection ValidateAndCreateConnection(string ownerUri)
+        {
+            ServerConnection serverConn = null;
+            if (ownerUri != null)
+            {
+                ConnectionInfo connInfo = null;
+                if (ConnectionServiceInstance.TryFindConnection(ownerUri, out connInfo))
+                {
+                    serverConn = ConnectionService.OpenServerConnection(connInfo);
+                }
+            }
+            return serverConn;
+        }
+
+        private ListRegisteredServersResult GetChildrenfromParentGroup(ServerGroup parentGroup)
+        {
+            var servers = parentGroup.RegisteredServers;
+            var groups = parentGroup.ServerGroups;
+
+            // Convert to appropriate variables and return
+            var serverResults = new List<RegisteredServerResult>();
+            foreach (var s in servers)
+            {
+                serverResults.Add(new RegisteredServerResult
+                {
+                    Name = s.Name,
+                    ServerName = s.ServerName,
+                    Description = s.Description,
+                    connectionDetails = ConnectionServiceInstance.ParseConnectionString(s.ConnectionString),
+                    RelativePath = s.KeyChain.Urn.SafeToString()
+                });
+            }
+
+            var groupsResults = new List<RegisteredServerGroup>();
+            foreach (var s in groups)
+            {
+                groupsResults.Add(new RegisteredServerGroup
+                {
+                    Name = s.Name,
+                    Description = s.Description,
+                    RelativePath = s.KeyChain.Urn.SafeToString()
+                });
+            }
+            ListRegisteredServersResult result = new ListRegisteredServersResult() { RegisteredServersList = serverResults, RegisteredServerGroups = groupsResults };
+            return result;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Internal for testing purposes only
+        /// </summary>
+        internal static ConnectionService ConnectionServiceInstance
```

Comment:
I think we've been putting properties above methods inside class definitions, at least that's what the .Net Framework coding guidelines recommend we do.